### PR TITLE
Dc mstp baud bug fix

### PR DESCRIPTION
**Pull request by dcmaverick**
Fix for not being able to reset default value,
JIRA: https://jira.se.com/browse/FIRMWARE-28880

### CODE REVIEW

File: ports/linux/rs485.c [hunk 1]
```diff
@@ -607,6 +607,13 @@ void RS485_Initialize(void)
             exit(EXIT_FAILURE);
         }
         /* if all goes well, set new divisor */
+        ioctl(RS485_Handle, TIOCSSERIAL, &newserial);
+    } else if(!RS485_SpecBaud && RS485_Baud == B38400) {
```
Comment (PeterJeaiem) [inline ports/linux/rs485.c:611]: !RS485_SpecBaud shouldnt be needed here since we are in the else clause of if (RS485_SpecBaud)

**Review by PeterJeaiem — CHANGES_REQUESTED** [review]

File: ports/linux/rs485.c (hunk 1 above)
Comment (dcmaverick) [reply ports/linux/rs485.c:611]: Very true, good eyes. thanks

**Review by PeterJeaiem — APPROVED** [review]

**Review by adrigonzo — APPROVED** [review]

File: ports/linux/rs485.c [hunk 2]
```diff
@@ -607,6 +607,13 @@ void RS485_Initialize(void)
             exit(EXIT_FAILURE);
         }
         /* if all goes well, set new divisor */
+        ioctl(RS485_Handle, TIOCSSERIAL, &newserial);
```
Comment (metelik) [inline ports/linux/rs485.c:610]: the ioctl is common for both if and else we could move it out conditional block

**Review by metelik — APPROVED** [review]

File: ports/linux/rs485.c (hunk 2 above)
Comment (dcmaverick) [reply ports/linux/rs485.c]: I've taken it outside of the if else statement

**Pull request merged**